mmdet/models/roi_heads/bbox_heads/bbox_head.py

```diff
@@ -317,122 +316,59 @@ def get_bboxes(self,
-               the length should be equal to the batch size.
+            img_shape (Sequence[int], optional): Maximum bounds for boxes,
+                specifies (H, W, C) or (H, W).
+            scale_factor (ndarray): Scale factor of the
+               image arrange as (w_scale, h_scale, w_scale, h_scale).
             rescale (bool): If True, return boxes in original image space.
                 Default: False.
             cfg (obj:`ConfigDict`): `test_cfg` of Bbox Head. Default: None
 
         Returns:
-            tuple[list[Tensor], list[Tensor]] or tuple[Tensor, Tensor]:
-                If the input has a batch dimension, the return value is
-                a tuple of the list. The first list contains the boxes of
-                the corresponding image in a batch, each tensor has the
-                shape (num_boxes, 5) and last dimension 5 represent
-                (tl_x, tl_y, br_x, br_y, score). Each Tensor in the second
-                list is the labels with shape (num_boxes, ). The length of
-                both lists should be equal to batch_size. Otherwise return
-                value is a tuple of two tensors, the first tensor is the
-                boxes with scores, the second tensor is the labels, both
-                have the same shape as the first case.
+            tuple[Tensor, Tensor]:
+                Fisrt tensor is `det_bboxes`, has the shape
+                (num_boxes, 5) and last
+                dimension 5 represent (tl_x, tl_y, br_x, br_y, score).
+                Second tensor is the labels with shape (num_boxes, ).
         """
 
-        # TODO: revert to single image inference
-
         # some loss (Seesaw loss..) may have custom activation
         if self.custom_cls_channels:
             scores = self.loss_cls.get_activation(cls_score)
         else:
             scores = F.softmax(
                 cls_score, dim=-1) if cls_score is not None else None
-
-        if rois.ndim == 2:
-            # e.g. AugTest, Cascade R-CNN, HTC, SCNet...
-            batch_mode = False
-            # add batch dimension
-            if scores is not None:
-                scores = scores.unsqueeze(0)
-            if bbox_pred is not None:
-                bbox_pred = bbox_pred.unsqueeze(0)
-            rois = rois.unsqueeze(0)
-
-            assert isinstance(scale_factor, np.ndarray)
-            scale_factor = (scale_factor, )
-
-        elif rois.ndim == 3:
-            # all input tensor have batch dimension
-            batch_mode = True
-            assert isinstance(scale_factor, tuple)
-        else:
-            raise NotImplementedError(f'Unexpect shape of roi {rois.shape}')
-
         # bbox_pred would be None in some detector when with_reg is False,
         # e.g. Grid R-CNN.
         if bbox_pred is not None:
             bboxes = self.bbox_coder.decode(
-                rois[..., 1:], bbox_pred, max_shape=img_shape)
+                rois[:, 1:], bbox_pred, max_shape=img_shape)
```

Comment:
Can we keep `rois[...,]` as this is more general?

Reply:
sure